packages/flutter_tools/test/integration.shard/flutter_build_config_only_test.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+// Test that configOnly creates the gradlew file and does not assemble and app.
+void main() {
+  late Directory tempDir;
+  late String flutterBin;
+  late Directory exampleAppDir;
+
+  setUp(() async {
+    tempDir = createResolvedTempDirectorySync('flutter_build_test.');
+    flutterBin = fileSystem.path.join(
+      getFlutterRoot(),
+      'bin',
+      'flutter',
+    );
+    exampleAppDir = tempDir.childDirectory('bbb').childDirectory('example');
+
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--template=plugin',
+      '--platforms=android',
+      'bbb',
+    ], workingDirectory: tempDir.path);
+  });
+
+  tearDown(() async {
+    tryToDelete(tempDir);
+  });
+
+  test(
+    'flutter build apk --configOnly should create gradlew and not assemble',
+    () async {
+      final File gradleFile = fileSystem
+          .directory(exampleAppDir)
+          .childDirectory('android')
+          .childFile('gradlew');
```

Comment:
will this be `gradlew.bat` on windows?

Reply:
yes it will. Is that something I need to account for in our tests?